addons/point_of_sale/static/src/app/store/pos_store.js

```diff
@@ -952,68 +956,84 @@ export class PosStore extends Reactive {
     cashierHasPriceControlRights() {
         return !this.config.restrict_price_control || this.get_cashier()._role == "manager";
     }
-    generate_unique_id() {
-        // Generates a public identification number for the order.
-        // The generated number must be unique and sequential. They are made 12 digit long
-        // to fit into EAN-13 barcodes, should it be needed
-
-        function zero_pad(num, size) {
-            var s = "" + num;
-            while (s.length < size) {
-                s = "0" + s;
-            }
-            return s;
-        }
-        return (
-            zero_pad(this.session.id, 5) +
-            "-" +
-            zero_pad(this.session.login_number, 3) +
-            "-" +
-            zero_pad(this.session.sequence_number, 4)
-        );
-    }
     createNewOrder(data = {}) {
+        if (!(data.pos_reference && data.sequence_number && data.tracking_number)) {
+            throw new Error("pos_reference, sequence_number and tracking_number are required");
+        }
+
         const fiscalPosition = this.models["account.fiscal.position"].find((fp) => {
             return fp.id === this.config.default_fiscal_position_id?.id;
         });
 
-        const uniqId = this.generate_unique_id();
         const order = this.models["pos.order"].create({
             session_id: this.session,
             company_id: this.company,
             config_id: this.config,
             picking_type_id: this.pickingType,
             user_id: this.user,
-            sequence_number: this.session.sequence_number,
             access_token: uuidv4(),
             ticket_code: random5Chars(),
             fiscal_position_id: fiscalPosition,
-            name: _t("Order %s", uniqId),
-            pos_reference: uniqId,
             ...data,
         });
 
-        this.session.sequence_number++;
         order.set_pricelist(this.config.pricelist_id);
+        order.recomputeOrderData();
```

Comment:
To manage the creation of the next reference, it may be preferable to do it here rather than in `add_new_order`.

You can simply call the new method that creates the reference by passing the order object. This way, there's no need to wait for it from here. 

The only method that will be asynchronous will be the method that retrieves the new reference; all the old order creation methods can remain synchrone.

Reply:
This also means that we have to bear in mind that the reference may take some time to be “ready”. 

But since no ticket will be printed directly, this is not a problem.

What do you think?

Reply:
I don't think this is a good suggestion. With it, when the `getNextOrderRefs` crashed, there will be an order that's invalid (no pos_reference, sequence_number and tracking_number). With the current solution, we prevent the creation when the order refs can't be fetched/generated.

Reply:
In this case the local fallback will create a reference ?